client/my-sites/upgrades/map-domain/index.jsx

```diff
@@ -1,148 +1,165 @@
+			selectedSite,
+			selectedSiteSlug,
+		} = this.props;
 
-	goBack: function() {
-		const selectedSite = this.props.sites && this.props.sites.getSelectedSite();
 		if ( ! selectedSite ) {
-			page( this.props.path.replace( '/mapping', '' ) );
+			page( paths.domainManagementRoot() );
 			return;
 		}
 
 		if ( selectedSite.is_vip ) {
-			page( paths.domainManagementList( selectedSite.slug ) );
+			page( paths.domainManagementList( selectedSiteSlug ) );
 			return;
 		}
 
-		page( '/domains/add/' + selectedSite.slug );
-	},
+		page( '/domains/add/' + selectedSiteSlug );
+	}
 
 	handleRegisterDomain( suggestion ) {
+		const { selectedSiteSlug } = this.props;
+
 		upgradesActions.addItem(
 			cartItems.domainRegistration( {
 				productSlug: suggestion.product_slug,
 				domain: suggestion.domain_name
 			} )
 		);
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + this.props.sites.getSelectedSite().slug );
-		}
-	},
+		page( '/checkout/' + selectedSiteSlug );
+	}
 
 	handleMapDomain( domain ) {
-		const selectedSite = this.props.sites.getSelectedSite();
+		const { selectedSite, selectedSiteSlug } = this.props;
 
 		this.setState( { errorMessage: null } );
 
 		// For VIP sites we handle domain mappings differently
 		// We don't go through the usual checkout process
 		// Instead, we add the mapping directly
 		if ( selectedSite.is_vip ) {
-			wpcom.addVipDomainMapping( selectedSite.ID, domain ).then( () => {
-				page( paths.domainManagementList( selectedSite.slug ) );
-			}, error => this.setState( { errorMessage: error.message } ) );
+			wpcom.addVipDomainMapping( selectedSite.ID, domain )
+				.then(
+					() => page( paths.domainManagementList( selectedSiteSlug ) ),
+					( error ) => this.setState( { errorMessage: error.message } )
+				);
 			return;
 		}
 
 		upgradesActions.addItem( cartItems.domainMapping( { domain } ) );
 
-		if ( this.isMounted() ) {
-			page( '/checkout/' + selectedSite.slug );
-		}
-	},
+		page( '/checkout/' + selectedSiteSlug );
+	}
 
-	render: function() {
-		let selectedSite;
+	render() {
+		const {
+			cart,
+			domainsWithPlansOnly,
+			initialQuery,
+			productsList,
+			selectedSite,
+			translate,
+		} = this.props;
 
-		if ( this.props.sites ) {
-			selectedSite = this.props.sites.getSelectedSite();
-		}
+		const {
+			errorMessage
+		} = this.state;
 
 		return (
 			<span>
 				<HeaderCake onClick={ this.goBack }>
-					{ this.translate( 'Map a Domain' ) }
+					{ translate( 'Map a Domain' ) }
 				</HeaderCake>
 
-				{ this.state.errorMessage && <Notice status="is-error" text={ this.state.errorMessage }/> }
+				{ errorMessage && <Notice status="is-error" text={ errorMessage } /> }
 
 				<MapDomainStep
-					{ ...omit( this.props, [ 'children', 'productsList', 'sites' ] ) }
-					products={ this.props.productsList.get() }
+					cart={ cart }
+					domainsWithPlansOnly={ domainsWithPlansOnly }
+					initialQuery={ initialQuery }
+					products={ productsList }
 					selectedSite={ selectedSite }
```

Comment:
Note that `selectedSite` is defined as `object|bool` in `MapDomainStep `, but this might have to be corrected - `getSelectedSite` can return `null`, but it won't return a `bool` value. We should check that with the rest of the components that receive this `selectedSite` along the way (starting with `DomainProductPrice` and `DomainRegistrationSuggestion`). We should be careful with updating these though, as they might be used in other components that might still pass a `bool` value.